File: 프로그래머스/unrated/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    idx = [x for x in range(len(arr))];
    if len(arr) % 2 == 1:
        return list(map(lambda x, i : x + n if i % 2 == 0 else x, arr, idx));
    else:
        return list(map(lambda x, i : x + n if i % 2 == 1 else x, arr, idx));

File: 프로그래머스/unrated/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
//     const len = arr.length
//     if(len % 2 === 1){
//         return arr.map((el, idx) => idx % 2 === 0 ? el + n : el)
//     }else{
//         return arr.map((el, idx) => idx % 2 === 1 ? el + n : el)
//     }
    
    return arr.map((el, idx) => arr.length % 2 === 1 ? (idx % 2 === 0 ? el + n : el) : (idx % 2 === 1 ? el + n : el))
}